R-packages/covidcast/R/utils.R

```diff
@@ -237,6 +237,85 @@ abbr_to_name = function(abbr, ignore.case = FALSE, perl = FALSE, fixed = FALSE,
               ties_method = ties_method)
 }
 
+#' Get FIPS codes from state abbreviations 
+#'
+#' Look up FIPS codes by state abbreviations (including District of Columbia and
+#' Puerto Rico); this function is based on `grep()`, and hence allows for
+#' regular expressions.  
+#'
+#' @param abbr Vector of state abbreviations to look up.
+#' @param ignore.case,perl,fixed Arguments to pass to `grep()`, with the same
+#'   defaults as in the latter function. Hence, by default, regular expressions
+#'   are used; to match against a fixed string (no regular expressions), set
+#'   `fixed = TRUE`.
+#' @param ties_method If "first", then only the first match for each name is
+#'   returned. If "all", then all matches for each name are returned.
+#'
+#' @return A vector of FIPS codes if `ties_method` equals "first", and a list of
+#'   FIPS codes otherwise. 
+#'
+#' @examples
+#' abbr_to_fips("PA")
+#' abbr_to_fips(c("PA", "PR", "DC"))
+#'
+#' # Note that name_to_fips() works for state names too:
+#' name_to_fips("^Pennsylvania$")
+#' 
+#' @seealso [abbr_to_name()]
+#' @export
+abbr_to_fips = function(abbr, ignore.case = FALSE, perl = FALSE, fixed = FALSE,
+                        ties_method = c("first", "all")) {
+  # First get rid of United States from state_census, then convert FIPS codes to
+  # appropriate character format 
+  df = state_census %>% dplyr::filter(STATE > 0) %>%
+    mutate(STATE = sprintf("%02d000", STATE))
```

Comment:
I feel like we want a `format_state_fips` function that does `sprintf("%02d000", STATE)`, since we do this formatting a lot. Saves us some copy-paste.

Reply:
Done.